src/main/java/com/wire/bots/hold/utils/HtmlGenerator.java

```diff
@@ -0,0 +1,73 @@
+package com.wire.bots.hold.utils;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import com.wire.xenon.tools.Logger;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public class HtmlGenerator {
+
+    private final static MustacheFactory mustacheFactory;
+
+    static {
+        mustacheFactory = new DefaultMustacheFactory();
+    }
+
+    private static Mustache indexTemplate() {
+        String path = "templates/index.html";
+        return mustacheFactory.compile(path);
+    }
+
+    private static Mustache eventsTemplate() {
+        String path = "templates/events.html";
+        return mustacheFactory.compile(path);
+    }
+
+    private static Mustache devicesTemplate() {
+        String path = "templates/devices.html";
+        return mustacheFactory.compile(path);
+    }
+
+    private static Mustache conversationTemplate() {
+        String path = "templates/conversation.html";
+        return mustacheFactory.compile(path);
+    }
+
+    public static String execute(Object model, TemplateType templateType) {
```

Comment:
A cool thing could have been creating an interface or abstract class like "Template" and then only the classes that we actually want to put in templates will implement that interface. This way the method here could accept a "Template" instead of any object.
Then the interface can expose a method like getTemplateType, so each class will declare which is the relevant type and we won't even need the switch here.

That's what I thought the other time when you showed me this, but it's just another refinement, the implementation you made already improves the duplicated code.